pytorch/training/docker/1.11/py3/cu113/Dockerfile.gpu

```diff
@@ -46,11 +46,11 @@ ENV TORCH_CUDA_ARCH_LIST="3.7 5.0 7.0+PTX 8.0"
 ENV TORCH_NVCC_FLAGS="-Xfatbin -compress-all"
 
 ENV CUDNN_VERSION=8.2.0.53
-ENV NCCL_VERSION=2.10.3
+ENV NCCL_VERSION=2.12.12
 ENV HOROVOD_VERSION=0.24.3
 ENV EFA_VERSION=1.15.2
 ENV OMPI_VERSION=4.1.1
-ENV BRANCH_OFI=1.2.0-aws
+ENV BRANCH_OFI=1.4.0-aws
```

Comment:
Hi @harishneit, let's change it to `aws` [branch](https://github.com/aws/aws-ofi-nccl/tree/aws), so it always pointing to the latest OFI_NCCL version.

Reply:
Relying on a proper release tag is safer as we can expect `aws-ofi-nccl` to do some QA before creating the release.  We can't expect the same with a branch.

Reply:
checked with Rashika from OFI_NCCL. the `aws` branch is a stable branch, and it is safe to use it. DLAMI also uses `aws` branch in its build process. https://github.com/aws/deep-learning-amis/blob/master/scripts/integrations/efa/build_efa_nccl_plugin.py#L41

I think we can follow the same in DLC

Reply:
Great! If we have that guarantee we could use the branch.

Reply:
But one issue will be the container builds will not be reproducible, do we want to guarantee that?

Reply:
like other unpinned packages, we can always use the latest stable version of OFI-NCCL. IMHO, the package itself should maintain the backward compatibility, performance and etc to make sure it's reproducible.

Reply:
As discussed offline, let's used the pinned version in this PR to preserve the reproducibility of container build.